pkg/virt-controller/watch/pool/pool.go

```diff
@@ -1264,32 +1346,36 @@ func (c *Controller) pruneUnusedRevisions(pool *poolv1.VirtualMachinePool, vms [
 }
 
 func (c *Controller) update(pool *poolv1.VirtualMachinePool, vms []*virtv1.VirtualMachine) (common.SyncError, bool) {
-	// List of VMs that need to be updated
-	vmOutdatedList := []*virtv1.VirtualMachine{}
-	// List of VMs that are up-to-date that need to be checked to see if VMI is up-to-date
-	vmUpdatedList := []*virtv1.VirtualMachine{}
+	var err error
+	filteredVms := vms
 
-	for _, vm := range vms {
-		outdated, err := c.isOutdatedVM(pool, vm)
-		if err != nil {
-			return common.NewSyncError(fmt.Errorf("error while detected outdated VMs: %v", err), FailedUpdateReason), false
-		}
+	if pool.Spec.UpdateStrategy != nil && pool.Spec.UpdateStrategy.Unmanaged != nil && !*pool.Spec.UpdateStrategy.Unmanaged {
+		log.Log.Infof("unmanaged update strategy is set, skipping update: updating VMs/VMIs is not allowed")
+		return nil, false
+	}
 
-		if outdated {
-			vmOutdatedList = append(vmOutdatedList, vm)
-		} else {
-			vmUpdatedList = append(vmUpdatedList, vm)
+	if hasOrderedSelectionPolicy(pool) {
```

Comment:
Shouldn't we use random sort by default, instead of doing nothing?

Reply:
We should distinguish between: `pool.Spec.UpdateStrategy == nil` and  `pool.Spec.UpdateStrategy.Proactive.SelectionPolicy == nil`.
In the latter, we should randomize since we say: 
```
type VirtualMachinePoolProactiveUpdateStrategy struct {
	// SelectionPolicy defines the priority in which VM instances are selected for proactive update
	// Defaults to "Random" base policy when no SelectionPolicy is configured
	// +optional
	SelectionPolicy *VirtualMachinePoolSelectionPolicy `json:"selectionPolicy,omitempty"`
}
```